### PR TITLE
Enable CDP builds

### PR DESCRIPTION
**Pull request by CyberDem0n**
and migrate to ubuntu18

### CODE REVIEW

File: delivery.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            version=$(egrep -m 1 "^${major_version/\./\\.}\." <<< "$ETCD_VERSIONS")
+
+            if [[ ! -z $prev ]]; then
+                IMAGE="registry-write.opensource.zalan.do/acid/etcd-cluster:$prev-$PATCH_VERSION"
+                docker build --build-arg ETCDVERSION_PREV=$version --build-arg ETCDVERSION=$prev -t $IMAGE .
+
+                # push docker images only for commits to the master branch
+                if [[ "x${CDP_SOURCE_BRANCH}" == "x" && "x${CDP_TARGET_BRANCH}" == "xmaster" ]]; then
+                    docker push $IMAGE
+                fi
```
Comment (mgomezch) [inline delivery.yaml]: This will still build the image but discard it if the pipeline triggers from a PR.  Is this desired?

File: delivery.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        apt-get update
+        apt-get install -y python3 python3-pip jq
+        pip3 install scm-source
+
+    - desc: Install Docker
+      cmd: 'curl -sSL https://get.docker.com/ | sh'
+
+    - desc: scm-source file create
+      cmd: |
+        scm-source
```
Comment (mgomezch) [inline delivery.yaml]: Per the [CDP docs](https://cloud.docs.zalando.net/concepts/builds/#scm-source), this should not be necessary:

> CDP will automatically inform Pierone of the SCM Source information on push. This means the /scm-source.json file does not need to be included in the pushed image.

File: delivery.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+build_steps:
+    - desc: Prepare Environment
+      cmd: |
+        apt-get update
+        apt-get install -y python3 python3-pip jq
```
Comment (mgomezch) [inline delivery.yaml]: Are Python 3 and pip necessary in the builder VM for this build?

File: delivery.yaml (hunk 1 above)
Comment (CyberDem0n) [reply delivery.yaml]: Yep, I want to check that build succeeds before merging PR.

File: delivery.yaml (hunk 2 above)
Comment (CyberDem0n) [reply delivery.yaml]: This is for compatibility, in case if I want to build and push it locally.

File: delivery.yaml (hunk 3 above)
Comment (CyberDem0n) [reply delivery.yaml]: They are needed only for scm-source

**Comment by Jan-M** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**